cmd/exit_handler.go

```diff
@@ -0,0 +1,312 @@
+package cmd
+
+import (
+	"net"
+	"os"
+	"os/signal"
+	"syscall"
+)
+
+// Priority is applied to callback or defer function
+type Priority int
+
+const (
+	Last Priority = iota
+	Indifferent
+)
+
+// SignalCallback_ callback function with priority
+type SignalCallback_ struct {
+	callbackFunc func()
+	priority     Priority
+}
+
+// NewSignalCallback creates a callback with priority
+func NewSignalCallback(callback func(), priority Priority) *SignalCallback_ {
+	return &SignalCallback_{
+		callback,
+		priority,
+	}
+}
+
+// GetPriority is a getter for callback's priority
+func (s *SignalCallback_) GetPriority() Priority {
+	return s.priority
+}
+
+// Call executes callback
+func (s *SignalCallback_) Call() {
+	s.callbackFunc()
+}
+
+// SignalHandler sends Signal to listeners and call registered callbacks
+type SignalHandler_ struct {
+	ch        chan os.Signal
+	listeners []net.Listener
+	callbacks []*SignalCallback_
+	signals   []os.Signal
+}
+
+// NewSignalHandler returns new SignalHandler registered for particular os.Signals
+func NewSignalHandler_(handledSignals []os.Signal) (*SignalHandler_, error) {
+	return &SignalHandler_{ch: make(chan os.Signal), signals: handledSignals}, nil
+}
+
+// AddListener to listeners list
+func (handler *SignalHandler_) AddListener(listener net.Listener) {
+	handler.listeners = append(handler.listeners, listener)
+}
+
+// GetChannel returns channel of os.Signal
+func (handler *SignalHandler_) GetChannel() chan os.Signal {
+	return handler.ch
+}
+
+// AddCallback to callbacks list
+func (handler *SignalHandler_) AddCallback(callback *SignalCallback_) {
+	handler.callbacks = append(handler.callbacks, callback)
+}
+
+// Register should be called as goroutine
+func (handler *SignalHandler_) Register() {
```

Comment:
After observing code execution a bit more, I realized that I added `SignalHandler_` while debugging process and actually it is not used))) so it can be removed. We have `ExitHandler` that is a replacement